discovery/discovery.go

```diff
@@ -40,8 +50,14 @@ func NewOrchestratorPool(node *core.LivepeerNode, addresses []string) *orchestra
 		glog.Error("Could not parse orchAddresses given - no URIs returned ")
 	}
 
+	var randomizedUris []*url.URL
+	r := rand.New(rand.NewSource(time.Now().Unix()))
+	for _, i := range r.Perm(len(uris)) {
+		uri := uris[i]
+		randomizedUris = append(randomizedUris, uri)
+	}
 	bcast := core.NewBroadcaster(node)
-	return &orchestratorPool{bcast: bcast, uris: uris}
+	return &orchestratorPool{bcast: bcast, uris: randomizedUris}
```

Comment:
I wonder how much randomizing the list will help in practice (for example, in the situation where a O is just a little bit faster at responding, it might still end being the selected O every time regardless of the order with which the goroutines for sending requests are created - I do suspect we'll have to implement the O concurrency cap to mitigate this as suggested by @j0sh). I guess trying it out as a quick fix on the test harness will answer that!

Reply:
Agree that this is a concern, esp when there aren't that many O's available. Will be interesting to see how often that might happen in practice with the `test-harness`